### DISCUSSION TITLE
Comment changer la couleur du fond d'un bouton en fonction d'un événement ?

### DISCUSSION THREAD

**Discussion by Mathis554** [witsa/synapps#751] [Q&A] [answered]
Bonjour,
Pouvais-vous m'aide, j'ai voulu changer la couleur d'un bouton en fonction du changement d'état de mon journal d'événement, mais je ne sais pas trop comment faire, j'ai crées des additionnelles dans mon acteur.
![image](https://github.com/user-attachments/assets/dc004024-a5a7-4829-873c-1d0edb0338f0)
Et j'ai relié le tout à la couleur de fond.
![image](https://github.com/user-attachments/assets/525ca019-a28a-4b7b-8527-5e963b6cedf8)
![image](https://github.com/user-attachments/assets/aa0e1e22-18f2-4875-a587-021eb69d2454)
![image](https://github.com/user-attachments/assets/246c802b-7b53-469e-901b-a76d1c43fb44)
Mais cela ne change pas la couleur alors j'ai essayé de changer le code par ce que j'ai trouvé dans le composite bouton clignotant.
![image](https://github.com/user-attachments/assets/e363b0f2-ca34-46d6-88eb-dc95d207d9c2)
Mais dans les deux cas ça m'affiche un message d'erreur
![image](https://github.com/user-attachments/assets/180f06c5-4d1b-4e19-a7c6-0cbc053993f6)

**Comment by manufitoussiwit** [comment, marked as answer]
Bonjour, 

Il faudrait mettre les deux couleurs sur le bouton. Ils seront plus facile à accéder.

```js
return context.value ? this.additionals.colorNOK :  this.additionals.colorOK;
```

Pour savoir comment accéder par script aux propriétés, aux additionnelles et aux paramètres de scène/composite, il suffit d'afficher les options de la propriété et tout en bas on peut même copier le chemin. Comme montré [ici](https://witsa.github.io/synapps/synapps-studio-releases/notes/1.2.0#copie-de-chemin-de-propri%C3%A9t%C3%A9additionnelle).

**Reply by Mathis554** [reply]
Merci, à vous deux, vous veniez de m'apprendre beaucoup de choses utiles merci. 😊
